tests/test_icpx.py

```diff
@@ -0,0 +1,309 @@
+
+    # verify that only icpx can be used, if it set as host and device compiler
+    # and has the same compiler version
+    def test_host_device_compiler_version(self):
+        valid_combs = [
+            [(ICPX, "2023.1"), (ICPX, "2023.1")],
+            [(ICPX, "2023.2"), (ICPX, "2023.2")],
+        ]
+
+        for comb in valid_combs:
+            self.assertTrue(
+                general_compiler_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                compiler_version_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                compiler_backend_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                software_dependency_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                full_filter_chain(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+
+        invalid_combs = [
+            [(ICPX, "2023.1"), (ICPX, "2023.2")],
+            [(ICPX, "2023.2"), (ICPX, "2023.1")],
+        ]
+
+        for comb in invalid_combs:
+            self.assertTrue(
+                general_compiler_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertFalse(
+                compiler_version_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                compiler_backend_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertTrue(
+                software_dependency_filter_typed(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+            self.assertFalse(
+                full_filter_chain(comb),
+                f"HOST_COMPILER: {comb[0][0]} {comb[0][1]}, "
+                f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}",
+            )
+
+    # Allowed back-ends are SYCL and the CPU back-ends
+    def test_host_device_compiler_backend(self):
+        valid_combs = [
+            [(ICPX, "2023.1"), (ICPX, "2023.1"), [(ALPAKA_ACC_SYCL_ENABLE, ON_VER)]],
```

Comment:
Just a design question. Does it make sense to use a version number instead simple on/off for the `ALPAKA_ACC_SYCL_ENABLE`? For `alpaka_ACC_GPU_CUDA_ENABLE` we use version numbers to combine `clang-cuda` with different CUDA SDK version.

`alpaka_ACC_GPU_HIP_ENABLE` also implement version numbers, but this is actual only homogeneous, because it is not possible to combine different HIP device compiler with different HIP SDKs.

If you cannot combine different sycl device compiler with differnt sycl SDKs, on/off is fine a make the usage easier. On the other side, it is a different behavior compare to the other two GPU platforms (I know sycl also works on CPU, but the major use is GPU).

Reply:
Hm, possibly in the future. Right now we only care about oneAPI. And you can't install `icpx` without also installing the SYCL runtime (or vice-versa) as far as I know. So it behaves like OpenMP: There is always a SYCL runtime bundled with the compiler.

Reply:
Okay, than we keep it, as it is. If it would change, we need to do an API break and release version 2.0.0 ;-)